### PR TITLE
[WiP] testing/ton-lite-client: new aport

### PR DESCRIPTION
**Pull request by OTLabs**
https://test.ton.org
Telegram Open Network (TON) lite client

### CODE REVIEW

**Comment by OTLabs** [comment]
Drone for aarch64 report following error:
```
cc1plus: error: unknown value 'armv8-a-march=armv8-a' for -march
--
229 | cc1plus: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a native
```

Other arm architectures build fine.

Please, suggest how to fix this error.

File: testing/ton-lite-client/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+package() {
+	cd $builddir/build
+	make DESTDIR=$pkgdir install
+
+	mkdir $pkgdir/usr/bin
+	rm -rf CMakeFiles
+	local i
+	for i in $(find . -executable -type f); do
+		cp $i $pkgdir/usr/bin/
+	done
```
Comment (Cogitri) [inline testing/ton-lite-client/APKBUILD]: Why doesn't the CMake build system do this?

File: testing/ton-lite-client/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Contributor: Oleg Titov <oleg.titov@gmail.com>
+# Maintainer: Oleg Titov <oleg.titov@gmail.com>
+
+pkgname=ton-lite-client
+pkgver=0.0.1
+pkgrel=0
+pkgdesc="Telegram Open Network (TON) lite client"
+url="https://test.ton.org"
+arch="all !aarch64"
```
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: Please, take a look at: https://github.com/alpinelinux/aports/pull/11196#issuecomment-529292098

File: testing/ton-lite-client/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+license="LGPL-2.0-or-later"
+options="!check" # No test suite from upstream
+makedepends="cmake gperf linux-headers openssl-dev zlib-dev"
+subpackages="$pkgname-doc"
+source="$pkgname-$pkgver.tar.xz::https://test.ton.org/ton-test-liteclient-full.tar.xz
+	https://test.ton.org/ton-lite-client-test1.config.json
+	"
```
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: Yep, not yet. It is in a very early stage. Please, take a lok at: https://test.ton.org/

File: testing/ton-lite-client/APKBUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cogitri) [reply testing/ton-lite-client/APKBUILD]: I did take a look, but it's unreasonable to expect other people which might interact with this APKBUILD in the future to read every PR for it to check why aarch64 is disabled

File: testing/ton-lite-client/APKBUILD (hunk 3 above)
[the remark this answers is not in the document]
Comment (Cogitri) [reply testing/ton-lite-client/APKBUILD]: Hm, maybe it's too early to include it in Alpine's main repos then :/

File: testing/ton-lite-client/APKBUILD (hunk 1 above)
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: Sorry, I have no answer. Must be for the very early stage of the project. Plane to give a closer look at CMake build system.

File: testing/ton-lite-client/APKBUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: What would be a solution? Include a comment in APKBUILD? What would be fine to mention in it?

File: testing/ton-lite-client/APKBUILD (hunk 3 above)
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: The expected release date is the end of September. So I am checking how the things are on non x86 architectures.

File: testing/ton-lite-client/APKBUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cogitri) [reply testing/ton-lite-client/APKBUILD]: # doesn't compile on aarch64 to to bad CFLAGS

File: testing/ton-lite-client/APKBUILD (hunk 3 above)
[the remark this answers is not in the document]
Comment (Cogitri) [reply testing/ton-lite-client/APKBUILD]: Okie, would it be OK for you if we put this on hold until then?

File: testing/ton-lite-client/APKBUILD (hunk 2 above)
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: Thank you!

File: testing/ton-lite-client/APKBUILD (hunk 3 above)
[the remark this answers is not in the document]
Comment (OTLabs) [reply testing/ton-lite-client/APKBUILD]: Yes, sure.

**Comment by OTLabs** [comment]
Source files are not available any more from original location, they have been moved to https://github.com/ton-blockchain/ton.

Closing this PR in waiting for first release at new location.

**Pull request closed without merging**